scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/component/parameter/defaults/SimpleLanguageDeterminer.scala

```diff
@@ -0,0 +1,36 @@
+package pl.touk.nussknacker.engine.definition.component.parameter.defaults
+
+import pl.touk.nussknacker.engine.api.definition._
+import pl.touk.nussknacker.engine.graph.expression.Expression
+
+trait SimpleLanguageDeterminer {
+
+  def calculateDefaultValue(
+      determinedEditor: Option[ParameterEditor],
+      defaultValue: Option[String]
+  ): Option[Expression] =
+    for {
+      value <- defaultValue
+      language <- determinedEditor
+        .collect {
+          case RawParameterEditor                   => Expression.Language.Spel
+          case simpleEditor: SimpleParameterEditor  => determineLanguageOf(simpleEditor)
+          case DualParameterEditor(simpleEditor, _) => determineLanguageOf(simpleEditor)
```

Comment:
```suggestion
          case DualParameterEditor(simpleEditor, DualEditorMode.SIMPLE) => determineLanguageOf(simpleEditor)
          case DualParameterEditor(_, DualEditorMode.RAW)               => Expression.Language.Spel
```

Reply:
nevermind, now that i think about it, changing the mode doesn't change the language, so this suggestion should be discarded